src/write.c

```diff
@@ -2591,21 +2591,25 @@ static avifResult avifEncoderWriteMiniBox(avifEncoder * encoder, avifRWStream *
         hasExplicitCodecTypes = AVIF_FALSE;
     }
 
-    uint32_t smallDimensionsFlag = image->width <= (1 << 7) && image->height <= (1 << 7);
-    const uint32_t codecConfigSize = 4; // 'av1C' always uses 4 bytes.
+    uint32_t largeDimensionsFlag = image->width > (1 << 7) || image->height > (1 << 7); // _minus1
+    const uint32_t codecConfigSize = 4;                                                 // 'av1C' always uses 4 bytes.
+    uint32_t alphaCodecConfigSize = 0;   // 0 if same codec config as main. Equal to codecConfigSize otherwise.
+    uint32_t gainmapCodecConfigSize = 0; // 0 if same codec config as main. Equal to codecConfigSize otherwise.
     uint32_t gainmapMetadataSize = 0;
-    const uint32_t fewCodecConfigBytesFlag = codecConfigSize < (1 << 3);
-    uint32_t fewItemDataBytesFlag = colorData->size <= (1 << 15) && (!alphaData || alphaData->size < (1 << 15));
-    uint32_t fewMetadataBytesFlag = image->icc.size <= (1 << 10) && image->exif.size <= (1 << 10) && image->xmp.size <= (1 << 10);
+    const uint32_t largeCodecConfigFlag = codecConfigSize >= (1 << 3);
+    uint32_t largeItemDataFlag = colorData->size > (1 << 15) // _minus1
+                                 || (alphaData && alphaData->size >= (1 << 15));
+    uint32_t largeMetadataFlag = image->icc.size > (1 << 10) || image->exif.size > (1 << 10) || image->xmp.size > (1 << 10); // _minus1
 
     if (hasGainmap) {
         AVIF_ASSERT_OR_RETURN(image->gainMap != NULL && image->gainMap->image != NULL);
         gainmapMetadataSize = avifGainMapMetadataSize(image->gainMap);
         AVIF_ASSERT_OR_RETURN(gainmapData != NULL);
 
-        smallDimensionsFlag &= image->gainMap->image->width <= (1 << 7) && image->gainMap->image->height <= (1 << 7);
-        fewItemDataBytesFlag &= gainmapData->size < (1 << 15);
-        fewMetadataBytesFlag &= encoder->data->altImageMetadata->icc.size <= (1 << 10) && gainmapMetadataSize <= (1 << 10);
+        largeDimensionsFlag &= image->gainMap->image->width > (1 << 7) || image->gainMap->image->height > (1 << 7); // _minus1
+        largeItemDataFlag &= gainmapData->size >= (1 << 15);
+        largeMetadataFlag &= encoder->data->altImageMetadata->icc.size > (1 << 10) // _minus1
+                             || gainmapMetadataSize >= (1 << 10);
```

Comment:
IMPORTANT: The original code is `gainmapMetadataSize <= (1 << 10)`. If we invert the condition, the new code should be `gainmapMetadataSize > (1 << 10)`.

Is the original code wrong?

Reply:
> Is the original code wrong?

Yes, or it was based on a draft that signaled `gainmap_metadata_size_minus1` which is no longer the case.